packages/block-library/src/preformatted/block.json

```diff
@@ -6,7 +6,8 @@
 			"type": "string",
 			"source": "html",
 			"selector": "pre",
-			"default": ""
+			"default": "",
+			"__unstablePreserveWhiteSpace": true
```

Comment:
Why is this in the block.json. Isn't this a RichText prop?

Reply:
It is needed for merging the values, which uses `create` and `toHTMLString` under the hood. It is similar to `__unstableMultilineWrapperTags`: https://github.com/WordPress/gutenberg/search?q=__unstableMultilineWrapperTags&unscoped_q=__unstableMultilineWrapperTags. I marked it unstable as maybe there's other ways we want to do this in future.

Reply:
mmm didn't know about that one. not great to be honest. Maybe the root issue is with the "merge" itself. maybe it should move to RichText somehow.

Reply:
Yes, we should have a look at it. The problem is that selection is stored based on the rich text value and not based on the HTML.

Reply:
Does it mean we should add information about these flags in the selection reducer?

Reply:
I'm not sure... Normally they are set with `RichText` props. We should document the selection reducers to say that the offsets are based on the rich text values, yes.

Reply:
I've added a bit more documentation for `WPSelectionObject`. I agree that the merge function could be improved, but this should be looked at separately while this PR fixes an issue. The attribute property has been marked unstable so it could be adjusted in the future.